Source/Core/DolphinQt2/HotkeyScheduler.cpp

```diff
@@ -0,0 +1,353 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "DolphinQt2/HotkeyScheduler.h"
+
+#include <algorithm>
+#include <thread>
+
+#include <QCoreApplication>
+
+#include "AudioCommon/AudioCommon.h"
+#include "Common/Thread.h"
+#include "Core/Core.h"
+#include "Core/HotkeyManager.h"
+#include "Core/IOS/IOS.h"
+#include "Core/IOS/USB/Bluetooth/BTBase.h"
+#include "Core/State.h"
+#include "DolphinQt2/MainWindow.h"
+#include "DolphinQt2/Settings.h"
+#include "InputCommon/ControllerInterface/ControllerInterface.h"
+#include "VideoCommon/VertexShaderManager.h"
+#include "VideoCommon/VideoConfig.h"
+
+constexpr const char* DUBOIS_ALGORITHM_SHADER = "dubois";
+
+HotkeyScheduler::HotkeyScheduler() : m_stop_requested(false)
+{
+  HotkeyManagerEmu::Initialize();
+  HotkeyManagerEmu::LoadConfig();
+  HotkeyManagerEmu::Enable(true);
+}
+
+void HotkeyScheduler::Start()
+{
+  m_stop_requested.Set(false);
+  m_thread = std::thread(&HotkeyScheduler::Run, this);
+}
+
+void HotkeyScheduler::Stop()
+{
+  m_stop_requested.Set(true);
```

Comment:
Needs:
```c++
if (m_thread.joinable())
  m_thread.join();
```

Also, this class should have a destructor which calls this method. `std::thread`s will abort the program if they get destructed without first having been joined :/